### PR TITLE
Added Collaboration information for Online Pro

### PR DESCRIPTION
**Pull request by ahardin-rh**
Preview Build: http://file.rdu.redhat.com/~ahardin/12072017/pro-collaboration/dev_guide/projects.html#project-collaboration-in-online-pro

### CODE REVIEW

**Comment by ahardin-rh** [comment]
@abhgupta @damemi @thrasher-redhat PTAL. Thanks!

File: dev_guide/projects.adoc [hunk 1]
```diff
@@ -194,3 +194,98 @@ To delete a project using the CLI:
...
++
+Now, when the collaborator user signs in to
+link:https://manage.openshift.com[manage.openshift.com], they will see a card to
+log in to the web console for the same cluster as your subscription and, if the
+collaborator was granted access to your projects on the cluster, they will have
+access just like any other user.
+
+[[project-collaboration-removing-collaborators]]
+=== Removing Collaborators
+
```
Comment (ahardin-rh) [inline dev_guide/projects.adoc]: @damemi @abhgupta @thrasher-redhat Perhaps more can be said about how to remove collaborators? What steps would the user follow specifically?

File: dev_guide/projects.adoc [hunk 2]
```diff
@@ -194,3 +194,98 @@ To delete a project using the CLI:
...
+. On the *Collaboration* page, enter the *Red Hat Login ID* for the user in the user
+name field and click *Add Collaborator*.
++
+You should now see the user listed under your collaborators, the time the user
+was added, and an option to remove the user from your subscription.
++
+This does not automatically grant the user any access to your projects. Access
+must be granted manually by the project owner using `oc policy` commands. For
+example, you can log in to the cluster through the CLI using your access token
```
Comment (ahardin-rh) [inline dev_guide/projects.adoc]: How is this specifically done via the web console vs. through the CLI?

Comment (abhgupta) [reply dev_guide/projects.adoc]: **Via the CLI:**
Add a role to a user: oc policy add-role-to-user \<role-name\> \<username\>
Remove a role from user: oc policy remove-role-from-user \<role-name\> \<username\>
Common examples for \<role-name\> could be view, edit, and admin
CLI Docs: https://docs.openshift.com/container-platform/3.7/admin_solutions/user_role_mgmt.html#control-user-roles

**Via the Console:**
Within a project, go to Resources --> Membership
From here you can add/remove roles to/from users

File: dev_guide/projects.adoc (hunk 1 above)
Comment (abhgupta) [reply dev_guide/projects.adoc]: In the registration application, when you see the list of collaborators you have added, you will see a "Remove" buttton next to each collaborator. Clicking the button will result in the collaborator being removed from your Pro tier subscription. The Pro plan owner is still responsible for removing any roles granted to the collaborator in their projects.

File: dev_guide/projects.adoc [hunk 3]
```diff
@@ -194,3 +194,98 @@ To delete a project using the CLI:
...
+====
+
+Collaboration is not available in {product-title} Starter.
+
+[[collaboration-restrictions]]
+=== Collaboration Restrictions
+
+Collaborators only have access to the specific projects (on clusters owned by
+the {product-title} Pro subscriber) that they are granted access to. They can
+not add projects of their own ownership. Collaborators can only add resources.
```
Comment (abhgupta) [inline dev_guide/projects.adoc]: > Collaborators can only add resources.

This is not entirely correct. We should say something on the lines of:
Collaborators can only access the resources within the projects that they have been granted access to. Also, their ability to view/edit/manage the project resources will depend on the specific role that they have been granted within the project.

File: dev_guide/projects.adoc (hunk 1 above)
Comment (abhgupta) [reply dev_guide/projects.adoc]: You have covered the details adequately (in my opinion) in your section below. Specifically, the Pro tier owner needs to revoke/remove any roles granted to the collaborator in their projects. Additionally, they should remove the user as a collaborator in the registration application.

**Comment by ahardin-rh** [comment]
@abhgupta Thanks for your guidance. This is now updated. PTAL.

File: dev_guide/projects.adoc [hunk 4]
```diff
@@ -194,3 +194,160 @@ To delete a project using the CLI:
...
+hosted with {product-title}. This allows multiple users to gain access to
+projects under a single subscription, without having to pay a monthly fee for
+every account.
+
+[NOTE]
+====
+Existing {product-title} Pro users do not need to be added as collaborators.
+====
+
+Collaboration is not available in {product-title} Starter.
```
Comment (sferich888) [inline dev_guide/projects.adoc]: This probably needs to be a warning or call out of some kind!

File: dev_guide/projects.adoc [hunk 5]
```diff
@@ -194,3 +194,160 @@ To delete a project using the CLI:
...
+within the project.
+
+[[project-collaboration-adding-collaborators]]
+=== Adding Collaborators
+{product-title} Pro subscribers can add collaborators by following these steps:
+
+. Each user you want to add as a collaborator must create a free account at
+ link:https://developers.redhat.com/[developers.redhat.com]. Once your
+ collaborator has confirmed their Red Hat Developers account, you can add them
+ to your subscription.
```
Comment (sferich888) [inline dev_guide/projects.adoc]: How does a customer do this? How can Support do this?

File: dev_guide/projects.adoc [hunk 6]
```diff
@@ -194,3 +194,160 @@ To delete a project using the CLI:
...
+. On the *Collaboration* page, enter the *Red Hat Login ID* for the user in the user
+name field and click *Add Collaborator*.
++
+You should now see the user listed under your collaborators, the time the user
+was added, and an option to remove the user from your subscription.
++
+This does not automatically grant the user any access to your projects. Access
+must be granted manually by the project owner using
+xref:granting-project-access-using-the-cli[`oc policy` commands] or the
+xref:granting-project-access-using-the-web-console[web console].
```
Comment (sferich888) [inline dev_guide/projects.adoc]: Should we point to https://access.redhat.com/solutions/3093811 as docs for how to do this?

Comment (sferich888) [reply dev_guide/projects.adoc]: Or is this KCS now deprecated by the next section[s] of docs?

File: dev_guide/projects.adoc [hunk 7]
```diff
@@ -194,3 +194,160 @@ To delete a project using the CLI:
...
+access just like any other user.
+
+[[project-collaboration-removing-collaborators]]
+=== Removing Collaborators
+
+If at any time you wish to remove the user as a collaborator from your
+subscription, you can do so on the same *Collaboration* page you used to add
+them. It is important to note, however, that this will not automatically remove
+any roles you have assigned the user in your projects. These will need to be
+manually deleted, or the user may still have access to your projects.
```
Comment (sferich888) [inline dev_guide/projects.adoc]: Should we link to https://access.redhat.com/solutions/3094651 as docs on how to do this?

Comment (sferich888) [reply dev_guide/projects.adoc]: Or is this KCS now deprecated by the next section[s] of docs?

**Review by sferich888 — APPROVED** [review]

File: dev_guide/projects.adoc (hunk 5 above)
Comment (thrasher-redhat) [reply dev_guide/projects.adoc]: The customer simply needs to have their collaborator create a Red Hat Developers account.  Going through the normal registration process and email verification.  Once they have a developer account, they'll be able to see their login ID (see below), which is what is needed to add them to the pro user's collaboration management page.

File: dev_guide/projects.adoc (hunk 6 above)
Comment (abhgupta) [reply dev_guide/projects.adoc]: @sferich888 Doesn't CEE already have tier-2 access to be able to access all resources within a user's projects on all Pro clusters?

File: dev_guide/projects.adoc (hunk 7 above)
Comment (abhgupta) [reply dev_guide/projects.adoc]: @sferich888 Doesn't CEE already have tier-2 access to be able to access all resources within a user's projects on all Pro clusters?

File: dev_guide/projects.adoc [hunk 8]
```diff
@@ -194,3 +194,160 @@ To delete a project using the CLI:
...
+ to your subscription.
+
+ . Each collaborator must sign into
+link:https://developers.redhat.com/[developers.redhat.com] and click on their
+name in the upper-right corner to access their account details. Make note of the
+*Red Hat Login ID* on this page, as it is the user name you will be required to
+enter in order to associate the collaborator with your subscription.
+
+. Sign in to link:https://manage.openshift.com[manage.openshift.com] and click on
+*Manage Subscription* under the cluster you want to add the collaborator to.
```
Comment (bmcelvee) [inline dev_guide/projects.adoc]: s/add the collaborator to/add the collaborator --maybe?

File: dev_guide/projects.adoc [hunk 9]
```diff
@@ -194,3 +194,160 @@ To delete a project using the CLI:
...
+=== Granting Project Access Using the Web Console
+
+After xref:project-collaboration-adding-collaborators[adding the collaborator]
+to your {product-title} Pro subscription, you can grant project access to the
+collaborator using the web console.
+
+. From within a project, click *Resources*, then *Membership*.
+
+. Add roles (for example, *view*, *edit*, or *manage*) to the user you want to
+grant access to.
```
Comment (bmcelvee) [inline dev_guide/projects.adoc]: s/grant access to/grant access --maybe?

File: dev_guide/projects.adoc [hunk 10]
```diff
@@ -194,3 +194,160 @@ To delete a project using the CLI:
...
+Existing {product-title} Pro users do not need to be added as collaborators.
+====
+
+Collaboration is not available in {product-title} Starter.
+
+[[collaboration-restrictions]]
+=== Collaboration Restrictions
+
+Collaborators can only access the resources within the projects that they have
+been granted access to. Also, their ability to view, edit, and manage the
```
Comment (bmcelvee) [inline dev_guide/projects.adoc]: s/granted access to/granted access --maybe?

**Comment by bmcelvee** [comment]
A few tiny things from me, otherwise looks good! ❄️

File: dev_guide/projects.adoc [hunk 11]
```diff
@@ -194,3 +194,160 @@ To delete a project using the CLI:
...
+. Log in to the cluster through the CLI using your access token.
+
+. Remove a role (for example, *view*, *edit*, or *manage*) from a specific
+collaborator using the same user name listed on the *Collaboration* page by
+running:
++
+----
+$ oc policy remove-role-from-user <role-name> <username>
+----
+endif::[]
```
Comment (vikram-redhat) [inline dev_guide/projects.adoc]: endif::openshift-online[]

File: dev_guide/projects.adoc (hunk 6 above)
Comment (ahardin-rh) [reply dev_guide/projects.adoc]: @sferich888 I think this PR deprecates that KCS

**Comment by ahardin-rh** [comment]
Thank you all for the feedback! This is updated. 

No revision history needed, as this will only appear in Online. We don't maintain revision histories in Online.

**Comment by ahardin-rh** [comment]
The Online docs were published via https://github.com/openshift/openshift-docs/pull/6738 
I am updating the milestone to Next Release so that the other branches are addressed in the next weekly release.
cc @vikram-redhat

**Pull request merged**